src/core/common/logging.cpp

```diff
@@ -297,6 +297,8 @@ void otDump(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aId, const
     size_t                        idLen = strlen(aId) + kFixedStringPart;
     ot::String<kStringLineLength> string;
 
+    VerifyOrExit(otLoggingGetLevel() >= aLogLevel);
```

Comment:
nit: We only need this checking when `OPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE` is used?